### PR TITLE
Document Object Model 

### PR DESCRIPTION
**Pull request by pashapushak1996**
### Document Object Model

[Demo](https://pashapushak1996.github.io/js-dom-kottans/) |
[Code base](https://github.com/pashapushak1996/js-dom-kottans/)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Comment by pashapushak1996** [comment]
I have completed self-check!

File: submissions/pashapushak1996/js-dom/index.js [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+};
+
+const menuLinkOnClick = () => {
+    const sideMenuItems = document.querySelectorAll('.side-menu__item');
+
+    const characterItems = document.querySelectorAll('.character');
+    sideMenuItems.forEach((item, index) => {
+        item.addEventListener('click', toggleClass(characterItems, index, 'character_hide'));
+
+        item.addEventListener('click', toggleClass(sideMenuItems, index, 'active', 'add'));
```
Comment (madmaxWMFU) [inline submissions/pashapushak1996/js-dom/index.js:150]: Add event listener one for all menus.
Before rendering all content add the property id to the dataset element.
After clicking, we check the link's id with dataset id content wrap.

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]
Good work. See comment.

File: submissions/pashapushak1996/js-dom/index.js (hunk 1 above)
Comment (pashapushak1996) [reply submissions/pashapushak1996/js-dom/index.js]: I have done it)

File: submissions/pashapushak1996/js-dom/index.js [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+                    target.parentNode.classList.add('active');
+                    character.classList.remove('character_hide');
+                } else {
+                    character.classList.add('character_hide');
+                    sideMenuItems[index].classList.remove('active');
+                }
+            });
+        });
+    });
+}
```
Comment (madmaxWMFU) [inline submissions/pashapushak1996/js-dom/index.js:162]:
```suggestion
const listPersone = Array.from(document.querySelectorAll('.character'));
const menuList = document.querySelector('.side-menu__list');
const sideMenuItems = Array.from(document.querySelectorAll('.side-menu__item'));

const hideActiveLink = () => sideMenuItems.forEach(link => link.classList.remove('active'));

const toggleMenuItem = (selectItem) => {
    listPersone.forEach(item => {
        if(item.dataset.id === selectItem.id) {
            item.classList.remove('character_hide');
        } else {
            item.classList.add('character_hide');
        }
    });
}

const menuLinkOnClick = ({target}) => {
    hideActiveLink();
    target.parentElement.classList.add('active');
    toggleMenuItem(target);
}

menuList.addEventListener('click', menuLinkOnClick);
```

File: submissions/pashapushak1996/js-dom/index.html [hunk 3]
```diff
@@ -0,0 +1,27 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <meta name="viewport"
+        content="width=device-width">
+  <meta http-equiv="X-UA-Compatible" content="ie=edge">
+  <link href="reset.css" rel="stylesheet">
+  <link href="main.css" rel="stylesheet">
+  <script src="index.js"></script>
```
Comment (madmaxWMFU) [inline submissions/pashapushak1996/js-dom/index.html:10]: Add this before the end of the tag `body`

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]

File: submissions/pashapushak1996/js-dom/index.html (hunk 3 above)
Comment (pashapushak1996) [reply submissions/pashapushak1996/js-dom/index.html]: Thank you)

**Comment by madmaxWMFU** [comment]
Change, because not work
```
const hideActiveLink = () => {
    const sideMenuItems = Array.from(document.querySelectorAll('.side-menu__item'));
    sideMenuItems.forEach(link => link.classList.remove('active'));
}

const toggleMenuItem = (selectItem) => {
    const listPersone = Array.from(document.querySelectorAll('.character'));
    listPersone.forEach(item => {
        if(item.dataset.id === selectItem.id) {
            item.classList.remove('character_hide');
        } else {
            item.classList.add('character_hide');
        }
    });
}
```

**Comment by pashapushak1996** [comment]
Thank you!

**Review by madmaxWMFU — APPROVED** [review]
Great, I approve of your work. Good luck with future tasks!

**Pull request merged**